blocks/embed/embed.js

```diff
@@ -0,0 +1,263 @@
+  if (sessionStorage.getItem(key)) {
+    const currentConsentSet = new Set(sessionStorage.getItem(key).split(','));
+    if (!currentConsentSet.has(value)) {
+      sessionStorage.setItem(key, `${Array.from(currentConsentSet).join(',')},${value}`);
+    }
+  } else {
+    sessionStorage.setItem(key, value);
+  }
+};
+
+/**
+ * Gets consent status for a given source.
+ * @function getConsent
+ * @memberof EmbedBlock
+ * @param {string} source - The source to check consent for.
+ * @param {string} [key='videoConsent'] - The key to retrieve consent from in sessionStorage.
+ * @returns {boolean} - The consent status for the given source.
+ */
+const getConsent = (source, key = 'videoConsent') => {
+  let output = false;
+  if (sessionStorage.getItem(key)) {
+    const consentSet = new Set(sessionStorage.getItem(key).split(','));
+    if (consentSet.has(source)) {
+      output = true;
+    }
+  }
+  return output;
+};
+
+/**
+ * Embeds a YouTube video.
+ * @function embedYoutube
+ * @memberof EmbedBlock
+ * @param {URL} url - The URL of the YouTube video.
+ * @param {boolean} [autoplay=true] - Whether to autoplay the video. Defaults to true.
+ * @returns {string} - The HTML code for embedding the YouTube video.
+ */
+const embedYoutube = (url, autoplay = true) => {
+  const usp = new URLSearchParams(url.search);
+  const suffix = autoplay ? '&muted=1&autoplay=1' : '';
+  let vid = usp.get('v') ? encodeURIComponent(usp.get('v')) : '';
+  const embed = url.pathname;
+  if (url.origin.includes('youtu.be')) {
+    [, vid] = url.pathname.split('/');
+  }
+
+  vid = decodeURI(`${vid}`).replace(/\\/g, '');
+
+  const embedHTML = `<div class="video-embed-container">
+      <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&v=${vid}${suffix}` : embed}"
+      class="video-embed-iframe"
+      allow="autoplay; fullscreen; picture-in-picture; encrypted-media; accelerometer; gyroscope; picture-in-picture"
+      allowfullscreen=""
+      scrolling="no"
+      title="Content from Youtube"
+      loading="lazy"></iframe>
+      </div>`;
+  return embedHTML;
+};
+
+/**
+ * Embeds a VideoJS Video.
+ * @function embedVideoJS
+ * @memberof EmbedBlock
+ * @param {URL} url - The URL of the VideoJS video.
+ * @param {boolean} autoplay - Whether autoplay is enabled or not.
+ * @returns {string} - The HTML for embedding the VideoJS video.
+ * @todo Validate whether this src URL pattern is really correct for all DGL videos (contact Pete Chamberlin)
+ * @todo Determine whether this is the best place to include the css and js lib file for videoJS
+ * @todo CSS styling pass needed to ensure the VideoJS player is brand compliant
+ */
+const embedVideoJS = (url, autoplay = true, poster = null) => {
+  // eslint-disable-next-line comma-dangle
+  const [, , vid,] = url.pathname.split('/');
+  const embedHTML = `
+    <link rel="stylesheet" href="/styles/video-js.min.css">
+    <link rel="stylesheet" href="/styles/videojs-sap.css">
```

Comment:
This should be loaded via the `loadCSS` helper into the HTML head, not into the body.

Reply:
Maybe you could extend the concept of the lib on the embed to be an array and then later on (see my other comment) just support both, js and css files in the lib array via loadCSS and loadScript (from aem.js) - respectively.

Reply:
I figured there was an EDS way of doing that.  I'll take a look.

Reply:
I moved the JS file path into the EMBEDS_CONFIG, then call loadScript as appropriate in the loadEmbed method.